src/Schema/Provider/PhpFileSchemaProvider.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Schema\Provider;
+
+use Cycle\ORM\Schema;
+use RuntimeException;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Yii\Cycle\Schema\Converter\SchemaToPHP;
+use Yiisoft\Yii\Cycle\Schema\SchemaProviderInterface;
+
+/**
+ * Be careful, using this class may be insecure.
+ */
+final class PhpFileSchemaProvider implements SchemaProviderInterface
+{
+    public const MODE_READ_AND_WRITE = 0;
+    public const MODE_WRITE_ONLY = 1;
+
+    private string $file = '';
+    private int $mode = self::MODE_READ_AND_WRITE;
+
+    private Aliases $aliases;
+
+    public function __construct(Aliases $aliases)
+    {
+        $this->aliases = $aliases;
+    }
+
+    public function withConfig(array $config): self
+    {
+        $new = clone $this;
+
+        // required option
+        if ($this->file === '' && !array_key_exists('file', $config)) {
+            throw new \InvalidArgumentException('The "file" parameter is required.');
+        }
+        $new->file = $this->aliases->get($config['file']);
+
+        $new->mode = $config['mode'] ?? $this->mode;
+
+        return $new;
+    }
+
+    public function read(): ?array
+    {
+        if ($this->mode === self::MODE_WRITE_ONLY) {
+            throw new RuntimeException(__CLASS__ . ' can not read schema.');
+        }
+        if (!is_file($this->file)) {
+            return null;
+        }
+        return include $this->file;
+    }
+
+    public function write(array $schema): bool
+    {
+        $content = (new SchemaToPHP(new Schema($schema)))->convert();
+        file_put_contents($this->file, $content);
```

Comment:
if you write php files that other processes might be reading, make sure to do proper locking on the file.
In highly frequented environment, that file may be half written while other process reads it.

See https://github.com/graphaware/neo4j-php-ogm/issues/122#issuecomment-350660050 for details on the possible problems.
And https://github.com/graphaware/neo4j-php-ogm/pull/174/files for a possible solution.